components/resources/quizzes/langQuiz/LangResults.tsx

```diff
@@ -3,22 +3,70 @@ import SkillifyNavbar from "../shared/SkillifyNavbar";
 
 type LangResultsProps = {
   onBackClick: () => void;
+  score: { [key: string]: number };
 };
 
-const LangResults = ({ onBackClick }: LangResultsProps) => {
+const LangResults = ({ onBackClick, score }: LangResultsProps) => {
+  const maxLangScore = (score: { [key: string]: number }) => {
```

Comment:
```
const getPreferredLanguageForQuizResults = (score: LanguageQuizResultsMap) => { ... }
```

Reply:
write a unit test in this PR that tests this function with various inputs

Reply:
this function should but move into the api folder

Reply:
Done!